fix: remove as self from document (#334)
openfga/openfga.dev: docs/content/configuration-language.mdx

:::

### The Direct Relationship Type Restrictions

> `[<type1>,<type2>]`, when used in the context of the <ProductConcept section="what-is-a-relation-definition" linkName="relation definition" /> allows <ProductConcept section="what-are-direct-and-implied-relationships" linkName="direct relationships" /> to the objects of these specified types. The absence of `[]` disallows these direct relationships.

Reviewer: ```suggestion
`[<string, <string>, ...]`, when used at the beginning of a <ProductConcept section="what-is-a-relation-definition" linkName="relation definition" /> allows <ProductConcept section="what-are-direct-and-implied-relationships" linkName="direct relationships" /> from the objects of these specified types. The strings can be of three formats:
- `<type>`: indicating that tuples relating objects of those types as users can be written (e.g. `group:marketing` can be related if `group` is in the type restrictions)
- `<type:*>`: indicating that a tuple relating all objects of that type can be written (e.g. `user:*` can be added if `user:*` is in the type restrictions)
- `<type>#<relation>`: indicating that tuples with sets of users that are related to an object of that type by that particular relation (e.g. `group:marketing#member` can be added if `group#member` is in the type restrictions)

If the direct relationship type restrictions are missing, then no tuple can be written relating other objects as this particular relation with objects of this type (direct relationships are disallowed).
```

:::info

> `[]` in the <ProductName format={ProductNameFormat.ShortForm}/> DSL translates to `this` in the <ProductName format={ProductNameFormat.ShortForm}/> API syntax.

Reviewer: ```suggestion
`[<type1>, <type2>, ...]` in the <ProductName format={ProductNameFormat.ShortForm}/> DSL translates to `this` in the <ProductName format={ProductNameFormat.ShortForm}/> API syntax.
```

:::

For example, let's take a closer look at the `team` type.

<AuthzModelSnippetViewer
  configuration={{

>     schema_version: '1.1',

Reviewer: This would result in incorrect JSON though.

Author: The JSON will still be correct (it is a snippet)

The result is 
```
{
  "type": "team",
  "relations": {
    "member": {
      "this": {}
    }
  },
  "metadata": {
    "relations": {
      "member": {
        "directly_related_user_types": [
          {
            "type": "user"
          },
          {
            "type": "user:*"
          },
          {
            "type": "team",
            "relation": "member"
          }
        ]
      }
    }
  }
}
```

<img width="2032" alt="Screen Shot 2023-01-06 at 5 41 52 PM" src="https://user-images.githubusercontent.com/10730463/211112152-c8ed7dbb-aa7d-4ba6-80ec-152e90000143.png">

Notice the component will also remove the schema version for JSON as well.

    },
    metadata: {
      relations: {
        member: { directly_related_user_types: [{ type: 'user' }, { type: 'user:*'}, { type: 'team', relation: 'member' }] },
      },
    },
  }} skipVersion={true}
/>

This `team` _<ProductConcept section="what-is-a-type-definition" linkName="type definition" />_ defines all the _<ProductConcept section="what-is-a-relation" linkName="relations" />_ that _<ProductConcept section="what-is-a-user" linkName="users" />_ can have with an _<ProductConcept section="what-is-an-object" linkName="object" />_ of _type_ `team`. In this case the _relation_ is: `member`.

> Due to the direct relationship type restrictions (`[user, team#member]`) being used, a user in the system can have a **<ProductConcept section="what-are-direct-and-implied-relationships" linkName="direct relationship" />** with the `team` type as a `member` for users of 

Reviewer: ```suggestion
Due to the direct relationship type restrictions (`[user, team#member]`) being used, a user in the system can have a **<ProductConcept section="what-are-direct-and-implied-relationships" linkName="direct relationship" />** with the `team` type as a `member` for objects of 
```

- type `user`

> - `user` <ProductConcept section="what-is-type-bound-public-access" linkName="type bound public access" />

Reviewer: ```suggestion
- the `user` <ProductConcept section="what-is-type-bound-public-access" linkName="type bound public access" /> (`user:*`)
```

> - <ProductConcept section="what-is-type-bound-public-access" linkName="type bound public access" /> of type `user`

Reviewer: Maybe `public access bound to the user type` when used here in this sentence, or `user type bound public access`

Author: I will use user type bound public access.  I prefer to keep `type bound public access` together as a term.

> - [usersets](./modeling/building-blocks/usersets.mdx) of type `team`

Reviewer: ```suggestion
- [usersets](./modeling/building-blocks/usersets.mdx) that have a `team` type and a `member` relation (e.g. `team:product#member`)
```

| Zanzibar           | <ProductName format={ProductNameFormat.ShortForm}/> JSON | <ProductName format={ProductNameFormat.ShortForm}/> DSL |
| :----------------- | :------------------------------------------------------- | :------------------------------------------------------ |

> | `this`             | `this`                                                   | `[]`                                                  |

Reviewer: Note that `[]` (as an empty set) is invalid. Maybe we can link to direct relationship type restrictions instead?

Author: Hmm...here, we directly refer to the syntax.  Maybe I will do something like `[<type1>,<type2>]` with link to the direct relationship type restrictions?
